frontend/pages/shelter/[id].tsx

```diff
@@ -24,9 +24,13 @@ const Home = () => {
 		if (!isValidShelterId(shelterId)) return;
 
 		const fetchPetData = async () => {
-			const res = await new PetApi().fetchShelterPets(shelterId);
-			setPetData(res);
-			setLoading(false);
+			try {
+				const res = await new PetApi().fetchShelterPets(shelterId);
+				setPetData(res);
+			} catch (e) {
```

Comment:
'setLoading(false)` here too. If the image doesn't get uploaded, it wont appear on the grid of images, but the loading status should be correct.

Reply:
did you highlight the wrong code? this page isn't for upload

Reply:
Not sure. it was included in the changes.